bigbluebutton-html5/imports/ui/components/audio/audio-modal/container.jsx

```diff
@@ -37,17 +47,29 @@ export default lockContextContainer(withModalMounter(withTracker(({ mountModal,
     }
   }
 
+  const canJoinAudio = canUserJoinAudio();
   return ({
+    canJoinAudio: canJoinAudio,
     closeModal: () => {
       if (!Service.isConnecting()) mountModal(null);
     },
     joinMicrophone: () => {
       const call = new Promise((resolve, reject) => {
+        if (!canJoinAudio) {
+          notifyUser(ERROR_MESSAGE, true, "mute");
+          reject(() => {
+            Service.exitAudio();
+          });
+          return;
+        }
+        else {
         if (skipCheck) {
           resolve(Service.joinMicrophone());
+          localStorage.setItem("VOICE_USER_ID", Auth.userID);
```

Comment:
This is being added to localStorage only in skip check. This should be added even when user connects audio for first time as well